borg/archiver.py

```diff
@@ -69,6 +72,11 @@ def with_repository(fake=False, invert_fake=False, create=False, lock=True, excl
     def decorator(method):
         @functools.wraps(method)
         def wrapper(self, args, **kwargs):
+            nonlocal compatibility
+            if create:
+                compatibility = REPO_OPERATION_NO_CHECK
+            if compatibility is None:
+                raise Exception("with_repository called without compatibility argument")
```

Comment:
the call to check_repository_compatibility() is still missing here (somewhere below), right?